### PR TITLE
Add AMD definitions

### PR DESCRIPTION
**Pull request by djessup**
I realise this plugin is no longer under active development, however it is still useful today. 

I've created a patch to include AMD module definitions to make it more compatible with modern deployments using RequireJS or similar without needing ugly shim code.

The functionality is completely unchanged, and should be 100% backwards compatible. The only difference is the plug defines itself as an AMD module in supported environments.

### CODE REVIEW

**Comment by BorisMoore** [comment]
Thanks - yes, makes sense to do this. I will try to verify and pull within a few days...

**Comment by djessup** [comment]
Thanks Boris, let me know if there is anything else I can do to help.

**Comment by BorisMoore** [comment]
Yes, sure I'll let you know. I didn't manage to get to this yet, but not forgotten. Hopefully soon. (If it is more than a week more, feel free to ping me :)

File: jquery.tmpl.js [hunk 1]
```diff
@@ -7,7 +7,18 @@
  * Dual licensed under the MIT or GPL Version 2 licenses.
  * http://jquery.org/license
  */
-(function( jQuery, undefined ){
+(function (factory) {
+	if (typeof define === 'function' && define.amd) {
+		// AMD
+		define(['jquery'], factory);
+	} else if (typeof exports === 'object') {
```
Comment (BorisMoore) [inline jquery.tmpl.js]: On the server (node.js) the DOM is  (generally) not available, and nor is jQuery, so  I think supporting CommonJS may be unnecessary. Do you agree?

File: jquery.tmpl.js [hunk 2]
```diff
@@ -7,7 +7,18 @@
...
+		// CommonJS
+		factory(require('jquery'));
+	} else {
+		// Browser globals
+		factory(jQuery);
+	}
+}(function( jQuery, undefined ){
```
Comment (BorisMoore) [inline jquery.tmpl.js]: When loaded by AMD loaders, I'm not sure we can be certain that the second parameter is undefined, so we should probably write `}(function(jQuery) {`

So the whole thing could look like this:

```
(function( factory ) {
	if (typeof define === 'function' && define.amd) {
		// Loading from AMD script loader. Register as an anonymous module.
		define( ['jquery'], factory );
	} else {
		// Browser using plain <script> tag
		factory( jQuery );
	}
}(function( jQuery ){
...
```

**Comment by BorisMoore** [comment]
I'm looking at AMD and Node.js support, as appropriate, in JsRender and JsViews, along with  the jquery-tmpl case, so I'm working on them all in parallel, for a consistent approach. For jquery-tmpl and JsViews we need jQuery and the DOM - so node.js on the server will not be a scenario. So I think we can remove CommonJS support other than for JsRender.  Does that make sense?

**Comment by djessup** [comment]
Sorry! I completely missed these messages.

The code in the PR is just standard boilerplate loader code and I can't see any reason why your modified version wouldn't work just fine. Would you like me to update the PR?

**Comment by BorisMoore** [comment]
Yes, you could do that. Thanks...

**Comment by djessup** [comment]
Ok, CommonJS has been removed from the factory method - sorry I've been so slow with updates! Let me know if there is anything else you think needs attention to merge.

**Comment by BorisMoore** [comment]
Thanks. I have merged your the changes locally, but need to do a sanity check / run tests, before pushing. Unfortunately I am heading out tomorrow for a trip abroad, and may not be able to complete the pull for the moment - until returning... If I can get to it sooner, I will...

**Comment by djessup** [comment]
No problem, I'm sure it can wait a little bit longer :-)

**Pull request merged**